src/lib/y2network/widgets/slave_items.rb

```diff
@@ -13,13 +13,15 @@ module SlaveItems
       # Builds content for slave configuration dialog (used e.g. when configuring
       # bond slaves) according the given list of item_ids (see LanItems::Items)
       #
-      # @param [Array<Fixnum>] item_ids           list of indexes into LanItems::Items
+      # @param [Array<String>] slaves             list of device names
       # @param [Array<String>] enslaved_ifaces    list of device names of already enslaved devices
-      def slave_items_from(item_ids, enslaved_ifaces)
-        raise ArgumentError, "no slave device defined" if item_ids.nil?
+      def slave_items_from(slaves, enslaved_ifaces)
+        raise ArgumentError, "no slave device defined" if slaves.nil?
```

Comment:
please use more meaningful error message like `slaves cannot be nil` and also maybe we need to check if slaves is not empty as it also does not make sense.

Reply:
(hopefully) improved